task/build-image-index/0.1/build-image-index.yaml

```diff
@@ -74,15 +80,29 @@ spec:
 
       sed -i 's/^\s*short-name-mode\s*=\s*.*/short-name-mode = "disabled"/' /etc/containers/registries.conf
 
+      if [[ $# -ne 1 && "$ALWAYS_BUILD_INDEX" != "true" ]]; then
+        echo "Skipping image index generation while supplying multiple image inputs is unsupported."
+        exit 2
+      fi
+
       image_manifests=""
       buildah manifest create "$IMAGE"
       for i in $@
       do
         TOADD="$i"
-        if [[ $(echo $i | tr -cd ":" | wc -c) == 2 ]]; then
+        if [[ $(echo "$i" | tr -cd ":" | wc -c) == 2 ]]; then
           #we need to remove the tag, and just reference the digest
           #as tag + digest is not supported
-          TOADD="$(echo $i | cut -d: -f1)@sha256:$(echo $i | cut -d: -f3)"
+          TOADD_REPOSITORY="$(echo "$i" | cut -d: -f1)"
+          TOADD_DIGEST="sha256:$(echo "$i" | cut -d: -f3)"
+          TOADD="${TOADD_REPOSITORY}@${TOADD_DIGEST}"
+        fi
+        if [[ "$ALWAYS_BUILD_INDEX" != "true" ]]; then
+          echo "Skipping image index generation. Returning results for $TOADD"
+          echo -n "${TOADD_REPOSITORY}" > "$(results.IMAGE_URL.path)"
+          echo -n "${TOADD_DIGEST}" > "$(results.IMAGE_DIGEST.path)"
+          echo -n "${TOADD}" > "$(results.IMAGES.path)"
```

Comment:
Now that I look at it again, I noticed the script only initializes `TOADD_REPOSITORY` and `TOADD_DIGEST` in the `if [[ $(echo "$i" | tr -cd ":" | wc -c) == 2 ]]` case. Other cases will be handled wrong (either cause a failure with un-initialized vars or use vars that were set earlier in the for loop)

Reply:
good catch! I'll fix that really quick.